### PR TITLE
fix "The value of the parameter is not used"

### PR DESCRIPTION
**Pull request by jukzi**
reduces warnings and removes dead code

### CODE REVIEW

File: apitools/org.eclipse.pde.api.tools.tests/src/org/eclipse/pde/api/tools/problems/tests/ApiProblemTests.java [hunk 1]
```diff
@@ -581,20 +584,7 @@ public void testGetHashCodeResourcePathNull() {
 		IApiProblem problem = ApiProblemFactory.newApiProblem(null, null, new String[] { "test1, test2, test3" }, null, null, 2, 2, 2, IApiProblem.CATEGORY_COMPATIBILITY, IElementDescriptor.FIELD, IApiProblem.ILLEGAL_IMPLEMENT, IDelta.ANNOTATION_DEFAULT_VALUE); //$NON-NLS-1$
 		assertNotNull("there should have been a new problem created", problem); //$NON-NLS-1$
 		assertEquals("the hashcode should be equal to the sum of: id, resourcepath.hashCode", //$NON-NLS-1$
-				problem.hashCode(), (problem.getId() + 0 + argumentsHashcode(new String[] { "test1, test2, test3" }))); //$NON-NLS-1$
+				problem.hashCode(), (problem.getId() + 0 + Arrays.hashCode(new String[] { "test1, test2, test3" }))); //$NON-NLS-1$
```
Comment (HannesWell) [inline apitools/org.eclipse.pde.api.tools.tests/src/org/eclipse/pde/api/tools/problems/tests/ApiProblemTests.java:587]: `Objects.hash()` uses var-args and you could save here and above the explicit creation of an array.
```suggestion
				problem.hashCode(), (problem.getId() + 0 + Objects.hash("test1, test2, test3"))); //$NON-NLS-1$
```

**Review by HannesWell — COMMENTED** [review]
Thanks for this clean-up.

File: apitools/org.eclipse.pde.api.tools.tests/src/org/eclipse/pde/api/tools/problems/tests/ApiProblemTests.java (hunk 1 above)
Comment (jukzi) [reply apitools/org.eclipse.pde.api.tools.tests/src/org/eclipse/pde/api/tools/problems/tests/ApiProblemTests.java]: nice

**Comment by jukzi** [comment]
unrelated fail on mac: 
`Error:  Failed to execute goal org.eclipse.tycho:tycho-eclipse-plugin:4.0.5-SNAPSHOT:eclipse-run (repo-to-runnable) on project org.eclipse.pde.build.tests: Error while executing platform: The launcher bundle org.eclipse.equinox.launcher_*.jar was not found in /Users/runner/work/eclipse.pde/eclipse.pde/build/org.eclipse.pde.build.tests/target/sdk-product/plugins, files in directory: `

**Pull request merged**